scripts/formValidation.js

```diff
@@ -0,0 +1,116 @@
+const email = document.getElementById('email');
+const password = document.querySelector('#pwd');
+const passwordConfirm = document.querySelector('#pwd-confirm');
+const username = document.getElementById('username');
+const submitButton = document.querySelector('.submit-button');
+
+document.documentElement.style.setProperty('--red', '#F74747');
+document.documentElement.style.setProperty('--gray400', '#9CA3AF');
+
+// 페이지 로드 시 버튼을 비활성화
+submitButton.disabled = true;
+submitButton.style.backgroundColor = 'var(--gray400)';
+
+function enableButton(button) {
+    button.disabled = false;
+    button.style.backgroundColor = 'var(--blue100)';
+}
+
+function disableButton(button) {
+    button.disabled = true;
+    button.style.backgroundColor = 'var(--gray400)';
+}
+
+function checkFormValidity() {
+    const isEmailValid = validateEmail(email.value) === '';
+    const isPasswordValid = validatePassword(password.value) === '';
+    const isPasswordConfirmValid = validatePasswordConfirm(passwordConfirm.value, password.value) === '';
+    const isUsernameValid = validateUsername(username.value) === '';
+
+    // 모든 입력값이 유효하면 버튼 활성화, 아니면 비활성화
+    if (isEmailValid && isPasswordValid && isPasswordConfirmValid && isUsernameValid) {
+        enableButton(submitButton);
+    } else {
+        disableButton(submitButton);
+    }
+}
+
+function validateInput(inputElement, messageElement, validateCallback) {
+    const inputValue = inputElement.value;
+    const validationMessage = validateCallback(inputValue);
+
+    if (validationMessage) {
+        messageElement.textContent = validationMessage;
+        inputElement.style.border = '1px solid var(--red)';
+        return false;
+    } else {
+        messageElement.textContent = '';
+        inputElement.style.border = 'none';
+        return true;
+    }
```

Comment:
❗️ 수정요청
if문에서 return문으로 조건 만족시 해당 함수가 조기 종료되므로 else 문은 없어도 됩니다~

```suggestion
    if (validationMessage) {
        messageElement.textContent = validationMessage;
        inputElement.style.border = '1px solid var(--red)';
        return false;
    }
    messageElement.textContent = '';
    inputElement.style.border = 'none';
    return true;
```